### DISCUSSION TITLE
re-use knobs

### DISCUSSION THREAD

**Discussion by pke** [storybookjs/react-native#389] [Q&A] [answered]
It seems one can not predefine knobs and re-use them in stories.

```ts

const fields = array("...")

storiesOf("Button")
  .add("some story", () => <Button fields={fields}/>)
  .add("other", ()  => <Button fields={fields}/>)
```

None of the stories will have fields knobs available in the UI

**Comment by dannyhw** [comment, marked as answer]
Pretty sure It is possible but you might want to create a function that returns the knob and then call it in the context of a story. It's problematic to call it from outside a story because it might get applied globally.

**Reply by pke** [reply]
Yup I just did. It does seem to be slower though?

**Reply by pke** [reply]
Can `parameters` be knobs?

**Reply by dannyhw** [reply]
Not sure, maybe? that's not the intended purpose but sometimes stuff just works I guess 😅 .
